src/metabase/api/setup.clj

```diff
@@ -38,21 +38,33 @@
   (su/with-api-error-message (s/constrained su/NonBlankString setup/token-match?)
     "Token does not match the setup token."))
 
+(def ^:dynamic ^:private *disallow-api-setup-after-first-user-is-created*
+  "We must not allow users to setup multiple super users once the first user is created. But tests still need to be able
+  to do that. This var is redef'd to false by certain tests to allow that."
+  true)
+
 (defn- setup-create-user! [{:keys [email first-name last-name password]}]
+  (when (and *disallow-api-setup-after-first-user-is-created* (setup/has-user-setup))
```

Comment:
the logic here seems weird. It doesn't feel like an override here. I know we can deMorgan our way into this but the logic could be more clear.
`(when (and (setup/has-user-setup) (not *allow-api-setup...*)` reads more like what my brain interprets it as. There's already a user setup and the override is not in place. It feels backwards to check the override before even the condition. I know its equivalent but my brain is struggling for some reason.

Reply:
Good point, its easy from my pov to focus on the override itself. But it makes more sense to highlight on the triggering condition (aka there is a user already).